visualization/app/codeCharta/ui/copyToClipboardButton/getFilenamesWithHighestMetrics.ts

```diff
@@ -0,0 +1,45 @@
+import { hierarchy } from "d3-hierarchy"
+import { CodeMapNode, NodeType } from "../../codeCharta.model"
+
+export function getFilenamesWithHighestMetrics(fileStates: CodeMapNode) {
+	return fileStates[0].file.map
+}
+
+export function parse(map: CodeMapNode): FileNamesByMetrics {
+	const fileNamesByMetrics = new Map<string, FileToValue[]>()
+	for (const { data } of hierarchy(map)) {
+		if (data.type === NodeType.FILE && data.attributes) {
+			for (const [key, value] of Object.entries(data.attributes)) {
+				maybeAddToMap(key, value, data.name, fileNamesByMetrics)
+			}
+		}
+	}
+	return new Map<string, string[]>()
+}
+
+type FileNamesByMetrics = Map<string, string[]>
+type FileToValue = { name: string; value: number }
+//sry Ruben
+function maybeAddToMap(key: string, value: number, fileName: string, map: Map<string, FileToValue[]>) {
+	const keyIsNew = !map.has(key)
+	if (keyIsNew) {
+		map.set(key, [{ name: fileName, value }])
+	} else {
+		const previousValues = map.get(key)
+		if (previousValues.length < 10) {
+			map.set(key, [...previousValues, { name: fileName, value }])
+		} else {
+			const minValue = Math.min(...previousValues.map(pair => pair.value), value)
```

Comment:
I think this would be more performant: I would suggest to first create a list of all files sorted by metric value. After that I would just slice the 10 files with highest metric values off and use it as the list to be copied to clipboard.
Currently you have to find the MIN value "NUMBER_OF_TOTAL_FILES"-times.

Reply:
At the moment we have for this method a runtime of approximately:

(For cc.json with 2391 and 1.5 MB)
 70 ms

(For cc.json with 13500 files and 20,7 MB)
400 ms